server.js

```diff
@@ -0,0 +1,291 @@
+          if (!err) {
+            req.setEncoding('utf8');
+            let allData = '';
+            req.on('data', (data) => {
+              allData += data;
+            });
+            req.on('end', function() {
+              const postData = qs.parse(allData);
+              const htmlTemplate = `<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <title>The Elements - ${postData.elementName}</title>
+    <link rel="stylesheet" href="/css/styles.css" />
+  </head>
+  <body>
+    <h1>${postData.elementName}</h1>
+    <h2>${postData.elementSymbol}</h2>
+    <h3>Atomic number ${postData.elementAtomicNumber}</h3>
+    <p>
+      ${postData.elementDescription}
+    </p>
+    <p><a href="/">back</a></p>
+  </body>
+</html>`;
+              fs.writeFile(`public/${postData.elementName.toLowerCase()}.html`, htmlTemplate, 'utf8', function(err) {
+                if (err) {
+                  return console.log(err);
+                }
+              });
+            });
+            resp.writeHead(200, {
+              'Content-Type': 'application/json',
+            });
+            resp.end(`{ "sucess" : true }`);
+            // file doesn't exist
+          } else {
+            resp.writeHead(500, {
+              'Content-Type': 'application/json',
+            });
+            resp.end(`{ "error" : resource public${req.url} does not exist" }`);
+          }
+        });
+      } else {
+        resp.writeHead(401, {
+          'Content-Type': 'application/json',
+          'WWW-Authenticate': `Basic realm="Secure Area"`,
+        });
+        resp.end(`<html><body>Not Authorized</body></html>`);
+      }
+    }
+  }
+
+  if (req.method === 'DELETE') {
+    const headerObj = req.headers;
+    const hasAuthHeader = headerObj.hasOwnProperty('authorization');
+    if (!hasAuthHeader) {
+      resp.writeHead(401, {
+        'Content-Type': 'application/json',
+        'WWW-Authenticate': `Basic realm="Secure Area"`,
+      });
+      resp.end(`<html><body>Not Authorized</body></html>`);
+    } else if (hasAuthHeader) {
+      const base64auth = headerObj.authorization.substring(6, headerObj.authorization.length);
+      const decodedAuth = new Buffer.from(base64auth, 'base64').toString();
+      const user = decodedAuth.substring(0, decodedAuth.indexOf(':'));
+      const password = decodedAuth.substring(decodedAuth.indexOf(':') + 1, decodedAuth.length);
+      if (userPasswords.hasOwnProperty(user) && userPasswords[user] === password) {
+        fs.access(`public${req.url}`, function(err) {
+          // file exists
+          if (!err) {
+            // delete file
+            fs.unlink(`public${req.url}`, (err) => {
+              if (err) {
+                throw err;
+              }
+              console.log(`public${req.url} was deleted`);
+            });
+            // remake index.html
+            //req.on('end', function() {
+            fs.readdir('./public', (err, files) => {
```

Comment:
Handle the err from fs.readdir and send a proper error back to the user with a status of 500